### ISSUE TITLE
Carriage returns are stripped out before being passed to the character handler

### ISSUE THREAD

**Issue by jpribele** [libexpat/libexpat#924]
`<foo>Blah \r\n Bar</foo>` just using \r\n here to show the characters visual assume the actual char values.

Assuming multi line string like this that has a carriage and new line line, Windows line endings. the character handler will skip the carriage. 

```
 case BT_CR:
   ptr += MINBPC(enc);
   if (ptr == end)
     return XML_TOK_TRAILING_CR;
   if (BYTE_TYPE(enc, ptr) == BT_LF)
     ptr += MINBPC(enc);
   *nextTokPtr = ptr;
   return XML_TOK_DATA_NEWLINE;
 ```
 
 ```
  case XML_TOK_DATA_NEWLINE:
   if (characterDataHandler) {
     XML_Char c = 0xA;
     characterDataHandler(handlerArg, &c, 1);
   }
  ```
  
There doesn't appear to be any way to get the carriage sent to any of the handlers.

I've have few reasons I want to keep the carriage return. Mostly its to keep all bytes the same after reading from disk

**Comment by hartwork** [comment]
Hi @jpribele the XML spec — Expat implements XML 1.0 fourth edition — has a say in this at https://www.w3.org/TR/2006/REC-xml-20060816/#sec-line-ends:

> To simplify the tasks of [applications](https://www.w3.org/TR/2006/REC-xml-20060816/#dt-app), the [XML processor](https://www.w3.org/TR/2006/REC-xml-20060816/#dt-xml-proc) MUST behave as if it normalized all line breaks in external parsed entities (including the document entity) on input, before parsing, by translating both the two-character sequence #xD #xA and any #xD that is not followed by #xA to a single #xA character.

I believe that explains (and demands) stripping the carriage return (decimal code 13) away. Does that answer your question?

**Comment by jpribele** [comment]
Yes. I've been able to work around the issue. Thanks for clearing it up though.
